### PR TITLE
Update the document regarding Open JDK and Oracle JDK

### PR DESCRIPTION
**Pull request by houshengbo**
Our documents need to be changed to point according to the usage of
Open JDK or Oracle JDK. This PR adds the instructions to install
either of them as the Java environment of OpenWhisk.

Partially-closes: apache/incubator-openwhisk-release#169

## Related issue and scope

- [x] I opened an issue to propose and discuss this change (#????)

## My changes affect the following components

- [ ] API
- [ ] Controller
- [ ] Message Bus (e.g., Kafka)
- [ ] Loadbalancer
- [ ] Invoker
- [ ] Intrinsic actions (e.g., sequences, conductors)
- [ ] Data stores (e.g., CouchDB)
- [ ] Tests
- [ ] Deployment
- [ ] CLI
- [ ] General tooling
- [ ] Documentation

## Types of changes

- [ ] Bug fix (generally a non-breaking change which closes an issue).
- [ ] Enhancement or new feature (adds new functionality).
- [ ] Breaking change (a bug fix or enhancement which changes existing behavior).

## Checklist:

- [x] I signed an [Apache CLA](https://github.com/apache/incubator-openwhisk/blob/master/CONTRIBUTING.md).
- [x] I reviewed the [style guides](https://github.com/apache/incubator-openwhisk/wiki/Contributing:-Git-guidelines#code-readiness) and followed the recommendations (Travis CI will check :).
- [ ] I added tests to cover my changes.
- [ ] My changes require further changes to the documentation.
- [ ] I updated the documentation where necessary.

### CODE REVIEW

File: tools/ubuntu-setup/java8.sh [hunk 1]
```diff
@@ -19,10 +19,20 @@
 set -e
 set -x
 
-if [ "$(lsb_release -cs)" == "trusty" ]; then
+JAVA_SOURCE=${1:-"open"}
+
+if [ "$JAVA_SOURCE" != "oracle" ] ; then
+    if [ "$(lsb_release -cs)" == "trusty" ]; then
+        sudo apt-get install -y software-properties-common python-software-properties
+        sudo add-apt-repository ppa:jonathonf/openjdk -y
```
Comment (mrutkows) [inline tools/ubuntu-setup/java8.sh]: Not sure if this PPA build is "trusted"... https://askubuntu.com/questions/464755/how-to-install-openjdk-8-on-14-04-lts ... perhaps we should discuss this on mailing list?

Comment (mrutkows) [reply tools/ubuntu-setup/java8.sh]: https://launchpad.net/~jonathonf/+archive/ubuntu/openjdk

Comment (houshengbo) [reply tools/ubuntu-setup/java8.sh]: sudo add-apt-repository ppa:openjdk-r/ppa -y should be able to work.

File: tools/macos/README.md [hunk 2]
```diff
@@ -54,6 +54,24 @@ sudo easy_install pip
...
+# install Open JDK 8
+brew install adoptopenjdk-openjdk8
+# install scala
+brew install scala
+# install pip
+sudo easy_install pip
+# install script prerequisites
+sudo -H pip install docker==2.2.1 ansible==2.5.2 jinja2==2.9.6 couchdb==1.1 httplib2==0.9.2 requests==2.10.0' | bash
+```
+
```
Comment (csantanapr) [inline tools/macos/README.md]: User just needs jdk 8 to be installed, this would be to run tests using gradle, and to build the jar for invoker and controller, we should not have duplicate instructions.
I think giving them one set with openjdk is fine in case they don't have java sdk install.

File: tools/vagrant/Vagrantfile [hunk 3]
```diff
@@ -87,7 +87,7 @@ Vagrant.configure('2') do |config|
       echo PATH=${PATH}:${HOME}/bin:${OPENWHISK_HOME}/tools/build >> /etc/environment
       cd ${OPENWHISK_HOME}
       cd tools/ubuntu-setup
-      su vagrant -c 'source all.sh'
+      su vagrant -c 'source all.sh oracle'
```
Comment (csantanapr) [inline tools/vagrant/Vagrantfile]: Why hardcode `oracle` here when the default every where else is `open`?

If you want to allow user to easily switch to use `oracle` make it an envirable variable they can do `JAVA_SOURCE=oracle ./hello`

File: tools/macos/README.md (hunk 2 above)
Comment (rabbah) [reply tools/macos/README.md]: I suggested this - leaving the existing docs for mac in place. Because that’s been something we’ve been using for a while and some are ok with the existing instructions.

Comment (csantanapr) [reply tools/macos/README.md]: I’m ok also leaving like it is, and maybe an mention that openjdk 8 could be use. 
What I don’t like is the double instructions
